core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/java/org/openjdk/jmc/flightrecorder/rules/jdk/util/DefaultIItemResultSet.java

```diff
@@ -59,19 +62,26 @@ final class DefaultIItemResultSet implements IItemResultSet {
 	private final Map<String, ColumnInfo> info;
 	private final ArrayList<Object[]> data = new ArrayList<>();
 	private int cursor = -1;
+	private ExecutorService exec;
 
-	DefaultIItemResultSet(IItemCollection items, IItemQuery query) {
+	DefaultIItemResultSet(IItemCollection items, IItemQuery query, int configuredTimeout) {
 		this.query = query;
 		attributes.addAll(query.getAttributes());
 		aggregators.addAll(query.getAggregators());
 		info = new HashMap<>(attributes.size() + aggregators.size());
+		exec = Executors.newWorkStealingPool();
 		initializeMetadata();
-		calculateData(items);
+		try {
+			calculateData(items, configuredTimeout);
+		} catch (InterruptedException e) {
+			Thread.currentThread().interrupt();
+		}
 	}
 
 	@SuppressWarnings({"unchecked", "rawtypes"})
-	private void calculateData(IItemCollection input) {
+	private void calculateData(IItemCollection input, int configuredTimeout) throws InterruptedException {
 		input = input.apply(query.getFilter());
+		final IItemCollection newInput = input;
```

Comment:
Very minor nit, but this could be moved to be inside the else block where it's used? It's not required for the first chunk of the if-statement.